src/main/java/ru/skypro/homework/service/impl/AdServiceImpl.java

```diff
@@ -137,12 +137,11 @@ public boolean patchAdPictureById(MultipartFile image, int adId, String username
     @Transactional
     @Override
     public boolean deleteAdById(int id, String username) {
-        AdEntity adFromRepository = adRepository.findById(id);
         UserEntity userPostedAd = adRepository.findById(id).getUserRelated();
         UserEntity authorizedUser = userRepository.findByUsername(username);
         Role authorizedUserRole = authorizedUser.getRole();
 
-        if (userPostedAd.equals(authorizedUser) || authorizedUserRole == Role.ADMIN) {
+        if (authorizedUserRole == Role.USER && userPostedAd.equals(authorizedUser) || authorizedUserRole == Role.ADMIN) {
```

Comment:
```suggestion
        if (userPostedAd.equals(authorizedUser) || authorizedUserRole == Role.ADMIN) {
```
проверять на юзера не нужно